### PR TITLE
ARGO-510 Implement API Call to create projects

### PR DESCRIPTION
**Pull request by kaggis**
### Goal
Implement API Call to create project(s)

### Implementation
- In store package:
 - [x] Implement QueryProject
 - [x] Implement InsertProject
 - [x] Update unit tests 
- [x] Add projects package
 - [x] Implement getProjectNameByUUID
 - [x] Impelemnt getProjectUUIDByName
 - [x] Implement Find Project (return one/all) 
 - [x] Implement CreateProject
 - [x] Implement HasProject 
 - [x] Implement ExportJSON for Project structures
 - [x] Add Unit tests
- In Handlers package:
 - [x] Implement ListOne Project call
 - [x] Implement ListAll Project call
 - [x] Implement Create Project call
- [x] Refactor topics
 - [x] Use projectUUID in models
 - [x] Update methods to use projectUUID
- [x] Refactor subscriptions
 - [x] Use projectUUID in models
 - [x] Update methods to use projectUUID
 - [x] Refactor pushers to be aware of projectUUID
- [x] Update swagger definition
- [x] Update docs

### CODE REVIEW

File: handlers.go [hunk 1]
```diff
@@ -142,6 +143,148 @@ func WrapAuthorize(hfn http.Handler, routeName string) http.HandlerFunc {
...
+	urlVars := mux.Vars(r)
+	urlProject := urlVars["project"]
+
+	// Grab context references
+	refStr := context.Get(r, "str").(stores.Store)
+
+	// Read POST JSON body
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		respondErr(w, 500, "Bad Request body", "BAD_REQUEST")
```
Comment (skanct) [inline handlers.go]: I believe this should be 400 Bad Request

File: handlers.go [hunk 2]
```diff
@@ -142,6 +143,148 @@ func WrapAuthorize(hfn http.Handler, routeName string) http.HandlerFunc {
...
+	// Get Result Object
+	res, err := projects.CreateProject(uuid, urlProject, created, user, postBody.Description, refStr)
+
+	if err != nil {
+		if err.Error() == "exists" {
+			respondErr(w, 409, "Project already exists", "ALREADY_EXISTS")
+			return
+		}
+
+		respondErr(w, 500, err.Error(), "INTERNAL")
```
Comment (skanct) [inline handlers.go]: Is it supposed to be INTERNAL, or INTERNAL SERVER ERROR ?

Comment (kaggis) [inline handlers.go]: It is supposed to be "INTERNAL"

**Comment by kaggis** [comment]
please review
/cc @themiszamani @skanct

File: subscriptions/subscription.go [hunk 3]
```diff
@@ -135,130 +141,159 @@ func (sl *Subscriptions) ExportJSON() (string, error) {
...
-		sl.List = append(sl.List, curSub)
+		result.List = append(result.List, curSub)
 	}
-
+	return result, err
 }
 
-// LoadPushSubs returns all subscriptions defined in store that have a push configuration
-func (sl *Subscriptions) LoadPushSubs(store stores.Store) {
+// // LoadFromStore returns all subscriptions defined in store
```
Comment (skanct) [inline subscriptions/subscription.go]: Why don't you remove this, if it is not needed?

File: subscriptions/subscription.go [hunk 4]
```diff
@@ -135,130 +141,159 @@ func (sl *Subscriptions) ExportJSON() (string, error) {
...
-		rp := RetryPolicy{sub.RetPolicy, sub.RetPeriod}
-		curSub.PushCfg = PushConfig{sub.PushEndpoint, rp}
-	}
-
-	sl.List = append(sl.List, curSub)
-	return nil
-
-}
+// // LoadOne loads one subscription
+// func (sl *Subscriptions) LoadOne(project string, subname string, store stores.Store) error {
```
Comment (skanct) [inline subscriptions/subscription.go]: Why don't you remove, if it is not needed any more?

File: subscriptions/subscription.go [hunk 5]
```diff
@@ -135,130 +141,159 @@ func (sl *Subscriptions) ExportJSON() (string, error) {
...
-	result := Subscriptions{}
-	for _, value := range sl.List {
-		if value.Project == project {
-			result.List = append(result.List, value)
-		}
-	}
-
-	return result
-}
+// // GetSubByName returns a specific topic
```
Comment (skanct) [inline subscriptions/subscription.go]: Why don't you remove it, if it is not needed any more?

File: stores/query_models.go [hunk 6]
```diff
@@ -1,8 +1,10 @@
 package stores
 
+import "time"
+
 // QSub are the results of the Qsub query
 type QSub struct {
-	Project      string `bson:"project"`
+	ProjectUUID  string `bson:"project_uuid"`
```
Comment (kaggis) [inline stores/query_models.go]: For the clients it's not a breaking change. For existing data: migration to a slightly updated schema is necessary

File: doc/v1/docs/api_projects.md [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+{
+ "name": "projects/EGI/topics/monitoring"
+}
+```
+
+### Errors
+Please refer to section [Errors](api_errors.md) to see all possible Errors
+
+
+## [DELETE] Manage Topics - Delete topic
```
Comment (kaggis) [inline doc/v1/docs/api_projects.md]: Indeed this is a mispasted line. The rest of project documentation will be added with the implementation of the Update,Delete requests

**Comment by kaggis** [comment]
- [x] Changed 500 Error Status from "INTERNAL" to "INTERNAL_SERVER_ERROR"

please review
@themiszamani @skanct

**Pull request merged**